### DISCUSSION TITLE
Roadmap for xcprivacy files on .NET iOS?

### DISCUSSION THREAD

**Discussion by borrrden** [dotnet/maui#19548] [Q&A]
Apple is going to quickly start rejecting non compliant apps once again as per the following:

https://developer.apple.com/documentation/bundleresources/privacy_manifest_files?language=objc

I'm wondering if the tooling is present yet for dealing with this new kinds of files.

**Comment by phillippschmedt** [comment]
This will be mandatory for all iOS apps starting in spring 2024.
How can we deal with this in .NET MAUI?

**Comment by phillippschmedt** [comment]
@davidortinau  Can we get an official statement on this topic? This is mandatory for all iOS apps soon. Thank you!

**Comment by jfversluis** [comment]
We're tracking the work for that here: https://github.com/xamarin/xamarin-macios/issues/20059

**Reply by FM1973** [reply]
Any news on this topic? May first isn´t too far away. We do need this for .net 7 as well. We currently can´t upgrade to .net 8.

**Comment by FM1973** [comment]
This is what I did.

I created a file PrivacyInfo.xcprivacy in Platforms/iOS - this is a xml file (like a plist).
Then I set the content of the file to the following (this contains the permissions .net maui needs - without any special permissions you might need):

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>NSPrivacyAccessedAPITypes</key>
    <array>
      <dict>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategoryFileTimestamp</string>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>C617.1</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategorySystemBootTime</string>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>35F9.1</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategoryDiskSpace</string>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>E174.1</string>
        </array>
      </dict>
      <dict>
        <key>NSPrivacyAccessedAPIType</key>
        <string>NSPrivacyAccessedAPICategoryUserDefaults</string>
        <key>NSPrivacyAccessedAPITypeReasons</key>
        <array>
          <string>CA92.1</string>
        </array>
      </dict>
    </array>
  </dict>
</plist>
```

Then in the .csproj file of the app I´ve set a BundleResource to include the file in the build like this:

```
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
  <BundleResource Include="Platforms\iOS\PrivacyInfo.xcprivacy" LogicalName="PrivacyInfo.xcprivacy" />
</ItemGroup>
```
This seems to work. Apple doesn´t complain anymore, Hope this helps someone.

**Reply by Perksey** [reply]
Thank you so much for this! Just got told off by TestFlight for this, assuming it's MAUI builtin stuff that's using these APIs as I'm not intentionally using them anywhere.
